### PR TITLE
Implement `frames_data` function

### PR DESCRIPTION
**Pull request by bartekpacia**
Please check whether I did it correctly. I cannot test it in the concentrator software because of [this](https://github.com/knei-knurow/ConcentratorTest/issues/4).

### CODE REVIEW

File: frames.c [hunk 1]
```diff
@@ -14,11 +14,7 @@ void frames_create(uint8_t* frame,
     frame[1] = header[1];
     frame[2] = data_len;
     frame[3] = '+';
-    // for (uint8_t i = 4; i < data_len; i++) {
-    // frame[i] = data[i - 4];
```
Comment (TK-A369) [inline frames.c:18]: Yes, this comments can be deleted.

File: frames.c [hunk 2]
```diff
@@ -28,8 +24,10 @@ void frames_header(uint8_t* frame, uint8_t* header) {
     header[1] = frame[1];
 }
 
-void frames_data(uint8_t frame, uint8_t frame_len, uint8_t* data) {
-    // TODO: implement
+void frames_data(uint8_t* frame, uint8_t frame_len, uint8_t* data) {
+    for (uint8_t i = 4; i < frame_len; i++) {
```
Comment (TK-A369) [inline frames.c]: I think that there is a bug. It counts from 4 to frame_len - 1, so in example:
frame: AT\x04+DATA#C
with frame_len=10, `i` will be: 4, 5, 6, 7, 8, 9 - so 6 repeats.
Also, I think that memcpy may be faster.
`memcpy(frame + 4, data, frame_len - 6)`
But I didn't tested it yet.

**Review by TK-A369 — CHANGES_REQUESTED** [review]
I found a bug in `frames.c`, in function `frames_data`.

File: frames.c (hunk 2 above)
Comment (bartekpacia) [reply frames.c]: Thanks for the catch. I used `memcpy(frame + 4, data, frame_len - 6)` but I also didn't test this.
We also need to demonstrate more (preferably all) of the functions available in this package in the `example.c` file. If you feel like testing the fix and adding more examples then... go ahead:)

**Comment by TK-A369** [comment]
I also suggest to change frames_data function's name to frames_read_data. It will be easier to understand.
I can do these changes.

**Review by TK-A369 — APPROVED** [review]
I made some changes. I also updated example.
I think that this pull request is ready to be merged.

**Pull request merged**